salt/pillar/__init__.py

```diff
@@ -460,20 +460,23 @@ def _minion_cache_path(self, minion_id):
         """
         return os.path.join(self.opts["cachedir"], "pillar_cache", minion_id)
 
-    def fetch_pillar(self):
+    def fetch_pillar(self, save_override=True):
```

Comment:
Why does `save_override` default to `True`? Why would we ever want to cache the pillar override?

Reply:
I wonder about this too, do we even need the `save_override` option?

Reply:
fetch_pillar isn't what does the saving,its return is what gets saved. and as such i wanted to make sure if it is being called elsewhere from some other source either in salt or third party module it retained the same functionality it had.

Reply:
i can remove that. it only gets called twice in salt. will simplify the test a little bit as i can remove the second check.

Reply:
I'm mainly just curious why `save_override=False` doesn't need to be applied to line 520 as well
https://github.com/whytewolf/salt/blob/pillar_cache/salt/pillar/__init__.py#L520

Reply:
that one missing is a mistake. I"ll talk with @dwoz this morning, see if we want to skip the backwards compatibility. if we do I"ll just strip out the save_override. otherwise I'll fix that line.

Reply:
on second thought, no one should be calling PillarCache directly anyway. I'll just strip out the save_override. save the headache of correcting it later.